scripts/publish.js

```diff
@@ -221,6 +222,29 @@ const createStarterDist = async (basename, isTypescript = false) => {
     if (file === "TS-README.md") {
       fs.renameSync(dest, path.join(dir.dist, name, "README.md"))
     }
+    // if we copied over the gatsby-node file, append type-conditional slices config
+    if (file === "gatsby-node.js") {
+      fs.appendFileSync(
+        dest,
+        `
+exports.createPages = ({ actions }) => {
```

Comment:
Just starting to look at this – why not include this in the source code? It'd mean a little bit of duplication, but I think this would be easy to miss in the publish script

Reply:
I also mentioned to Khaled already that I think this needs to change as the current `gatsby-node.js` also has this code already. So with the publish script the resulting file will have it twice (which is incorrect).

@aghreed added that

Reply:
Ah, this is switching the file extension below, so this makes sense. I don't _think_ the root level `gatsby-node.js` is used for anything beyond local dev in this repo, but you can confirm with a dry run and looking at the output files before publishing https://github.com/gatsbyjs/homepage-starters/blob/main/package.json#L17

Reply:
Precisely @jxnblk — we copy over each plugin's `gatsby-node` not the root one. And this way we avoid the file extension problem. Original comment I left is here: https://github.com/gatsbyjs/homepage-starters/pull/148#discussion_r1016154508